src/main/java/seedu/address/model/supplyItem/SupplyItem.java

```diff
@@ -0,0 +1,114 @@
+package seedu.address.model.supplyItem;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Price;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Supply Item in Salesy
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class SupplyItem {
+
+    // Identity fields
+    private final String name;
```

Comment:
Should the name of the SupplyItem be tied to the Item field of the Supplier of this SupplyItem? 

If it is:
- Cannot add new SupplyItem if it's not being supplied by any supplier
- Editing a SupplyItem's name will edit the item field in corresponding supplier
- Editing a supplier's Item field will edit the SupplyItem it is supplying

If it is not:
- Possibility of SupplyItem existing with no supplier supplying it
- Possibility of SupplyItem's name and Item field of supplier being different

Reply:
@Bahamas20

Reply:
Depends on how you all want to design it. 

A. The most naive method will be to check if the `Person`'s `Item` is same as `name`

I feel for correctness, we should do A + scan AddressBook's list to ensure that `Person` matches the supplier in `SupplyItem`

Reply:
But this will be another feature, lets not clog up this pull request given its already huge.

Reply:
Yea makes sense but for both possibilities we can just check if such supplier exist and match before adding that supplyitem abit of a hassle but better than tying to item i feel